PHPCI/Command/RunCommand.php

```diff
@@ -65,89 +54,116 @@ protected function configure()
+        $this->logger->addInfo(Lang::get('finished_processing_builds'));
+    }
 
-            try {
-                // Logging relevant to this build should be stored
-                // against the build itself.
-                $buildDbLog = new BuildDBLogHandler($build, Logger::INFO);
-                $this->logger->pushHandler($buildDbLog);
+    /**
+     * Find a pending build for a project which has no running builds.
+     *
+     * @return Build|null
+     */
+    protected function findNextBuild()
+    {
+        $running = $this->validateRunningBuilds();
 
-                $builder = new Builder($build, $this->logger);
-                $builder->execute();
+        $this->logger->addInfo(Lang::get('finding_builds'));
+        $result = Factory::getStore('Build')->getByStatus(Build::STATUS_NEW);
+        $this->logger->addInfo(Lang::get('found_n_builds', count($result['items'])));
 
-                // After execution we no longer want to record the information
-                // back to this specific build so the handler should be removed.
-                $this->logger->popHandler($buildDbLog);
-            } catch (\Exception $ex) {
-                $build->setStatus(Build::STATUS_FAILED);
-                $build->setFinished(new \DateTime());
-                $build->setLog($build->getLog() . PHP_EOL . PHP_EOL . $ex->getMessage());
-                $store->save($build);
+        foreach ($result['items'] as $build) {
+            if (isset($running[$build->getProjectId()])) {
+                $this->logger->addInfo(Lang::get('skipping_build', $build->getId()));
+            } else {
+                return $build;
             }
-
         }
 
-        $this->logger->addInfo(Lang::get('finished_processing_builds'));
-
-        return $builds;
+        return null;
     }
 
-    public function setMaxBuilds($numBuilds)
+    /**
+     * Runs one build and cleans up after it finishes.
+     *
+     * @param Build $build
+     */
+    protected function runBuild(Build $build)
     {
-        $this->maxBuilds = (int)$numBuilds;
-    }
+        $build = BuildFactory::getBuild($build);
+        $this->logger->addInfo("Running build ".$build->getId());
 
-    public function setDaemon($fromDaemon)
-    {
-        $this->isFromDaemon = (bool)$fromDaemon;
+        $buildDbLog = new BuildDBLogHandler($build, Logger::INFO);
+        $this->logger->pushHandler($buildDbLog);
+
+        try {
+            $builder = new Builder($build, $this->logger);
+            $builder->execute();
+
+        } catch (Exception $ex) {
+            $build->setStatus(Build::STATUS_FAILED);
+            $build->setFinished(new DateTime());
+            $build->setLog($build->getLog() . PHP_EOL . PHP_EOL . $ex->getMessage());
+            Factory::getStore('Build')->save($build);
+        }
+
+        $this->removeBuildDirectory($build);
+        $this->logger->popHandler($buildDbLog);
+        $this->logger->addInfo("Build ".$build->getId()." ended");
```

Comment:
Please use sprintf for string formatting, this will help the translation step.

```php
        $this->logger->addInfo(sprintf("Build %d ended", $build->getId()));
```